### PR TITLE
Feature/ds 87 implement event listener consumer for notification service

### PR DESCRIPTION
**Pull request by M-Jhay**

### CODE REVIEW

File: src/DwitTech.NotificationService.Core/Events/EmailEventConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            _consumer = consumer;
+            _emailConsumerService = emailConsumerService;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            try
+            {
+                using var scope = _serviceScopeFactory.CreateScope();
+                var cancellationToken = CancellationTokenSource.CreateLinkedTokenSource(stoppingToken, _cancellationTokenSource.Token).Token;
```
Comment (giftedGeorge) [inline src/DwitTech.NotificationService.Core/Events/EmailEventConsumer.cs:40]: stoppingToken sounds like an action. Can you name it something else?

File: src/DwitTech.NotificationService.Core/Services/EmailConsumerService.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                await _emailService.SendEmail(emailDto);
+            }
+            catch (OperationCanceledException)
+            {
+                _logger.LogInformation("Email event consumption was canceled.");
+                throw;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred while consuming Kafka message");
```
Comment (giftedGeorge) [inline src/DwitTech.NotificationService.Core/Services/EmailConsumerService.cs]: Why is this exception considered an error, but the previous exception is considered an "information". We should maintain consistency.

File: src/DwitTech.NotificationService.WebApi/Properties/launchSettings.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
       "launchBrowser": true,
       "launchUrl": "swagger",
       "environmentVariables": {
-        "ASPNETCORE_ENVIRONMENT": "Local"
+        "ASPNETCORE_ENVIRONMENT": "Development"
```
Comment (giftedGeorge) [inline src/DwitTech.NotificationService.WebApi/Properties/launchSettings.json]: I think you forgot to return this to local

File: src/DwitTech.NotificationService.Core/Models/EmailSentEvent.cs
Comment (giftedGeorge) [inline src/DwitTech.NotificationService.Core/Models/EmailSentEvent.cs:1]: Looks like this class is never used. Please remove.

File: tests/DwitTech.NotificationService.Core.Tests/Events/EmailEventConsumerTest.cs
Comment (giftedGeorge) [inline tests/DwitTech.NotificationService.Core.Tests/Events/EmailEventConsumerTest.cs:1]: Why's this file commented out?

**Review by giftedGeorge — CHANGES_REQUESTED** [review]
Please check the comments for changes that may be relevant.

File: tests/DwitTech.NotificationService.Core.Tests/Events/EmailEventConsumerTest.cs
Comment (johnny2565) [inline tests/DwitTech.NotificationService.Core.Tests/Events/EmailEventConsumerTest.cs:1]: Please, kindly remove the file if has no place anymore

**Review by johnny2565 — CHANGES_REQUESTED** [review]
Please fix the comments on the PR

File: src/DwitTech.NotificationService.Core/Events/EmailEventConsumer.cs (hunk 1 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Core/Events/EmailEventConsumer.cs:40]: > stoppingToken

StoppingToken is a compound noun not an action word(verb)

File: tests/DwitTech.NotificationService.Core.Tests/Events/EmailEventConsumerTest.cs
Comment (M-Jhay) [reply tests/DwitTech.NotificationService.Core.Tests/Events/EmailEventConsumerTest.cs:1]: > Please, kindly remove the file if has no place anymore

This is outdated already.

Comment (M-Jhay) [reply tests/DwitTech.NotificationService.Core.Tests/Events/EmailEventConsumerTest.cs:1]: > Why's this file commented out?

This is outdated already. I already took the file out

File: src/DwitTech.NotificationService.Core/Services/EmailConsumerService.cs (hunk 2 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Core/Services/EmailConsumerService.cs]: > Why is this exception considered an error, but the previous exception is considered an "information". We should maintain consistency.

Resolved

File: src/DwitTech.NotificationService.WebApi/Properties/launchSettings.json (hunk 3 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.WebApi/Properties/launchSettings.json]: > Member

Resolved

File: src/DwitTech.NotificationService.Core/Models/EmailSentEvent.cs
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Core/Models/EmailSentEvent.cs:1]: > Looks like this class is never used. Please remove.

Resolved

**Comment by M-Jhay** [comment]
> 

Resolved

**Review by johnny2565 — APPROVED** [review]
The PR looks good to me

**Review by giftedGeorge — APPROVED** [review]

**Review by dabensymon — APPROVED** [review]

**Pull request merged**